### ISSUE TITLE
Typescript Issue on Function Parameter. 

### ISSUE THREAD

**Issue by kuasha420** [bameyrick/throttle-typescript#15]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `throttle-typescript@1.0.1` for the project I'm working on.

This will allow any function to be used and produce correct results 

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/throttle-typescript/dist/index.d.ts b/node_modules/throttle-typescript/dist/index.d.ts
index 06bbdaa..39352b0 100644
--- a/node_modules/throttle-typescript/dist/index.d.ts
+++ b/node_modules/throttle-typescript/dist/index.d.ts
@@ -1 +1 @@
-export declare function throttle(func: Function, limit: number): Function;
+export declare function throttle<T extends Function>(func: T, limit: number): T;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by bameyrick** [comment]
Could you provide an example of how you are using the throttle function that requires a return type so that I can create a test case to cover it please?

**Comment by kuasha420** [comment]
Hello, This is the Code. 

```ts
const onRecStart = useCallback(async () => {
    setRecording(true);
    await voice.startRecorder();
    const setRec = (e: RecordBackType) => setSecs(voice.mmssss(Math.floor(e.currentPosition)));
    voice.addRecordBackListener(throttle(setRec, 500));
  }, []);
```

Thank you for responding <3

**Comment by bameyrick** [comment]
This should hopefully be resolved for you in release `1.1.0`
